src/components.test.ts

```diff
@@ -0,0 +1,107 @@
+import * as F from "./figma-json";
+// import components from "./components";
+import defaultLayers from "./figma-default-layers";
+import { dump } from ".";
+
+// Inject a fake figma object into the global scope as a hack
+beforeAll(() => {
+  (globalThis as any).figma = {};
+});
+
+test("Takes components in the document and produces component ids like the REST api in the result", async () => {
+  const sourceComponent = {
```

Comment:
I get why we're not explicitly telling TS what type this is/totally OK, but it feels a little fragile to manually turn a Frame into a component and just hope we added the right keys.

Reply:
Agreed, do you have a better idea though?
It can't be a full-on figma `ComponentNode`, since I don't implement any methods. I think down the line a good figma simulation / mock would be ideal here.